src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -3477,4 +3477,35 @@ WindowID MessageHelper::ExtractWindowIdFromSmartObject(
   return mobile_apis::PredefinedWindows::DEFAULT_WINDOW;
 }
 
+void MessageHelper::RemoveEmptyMessageParams(
+    smart_objects::SmartObject& msg_params) {
+  if (msg_params.empty()) {
+    return;
+  }
+
+  std::set<std::string> keys = msg_params.enumerate();
+  std::set<std::string>::const_iterator iter_set = keys.begin();
+
+  for (; iter_set != keys.end(); ++iter_set) {
+    auto& param = msg_params[*iter_set];
+
+    if (smart_objects::SmartType_Array == param.getType()) {
+      smart_objects::SmartArray* array = param.asArray();
+
+      if (array == nullptr)
+        continue;
+
+      smart_objects::SmartArray::iterator iter_array = std::find_if(
+          array->begin(), array->end(), [](smart_objects::SmartObject item) {
+            if (smart_objects::SmartType_Array == item.getType() ||
+                smart_objects::SmartType_Map == item.getType())
+              return item.empty();
+            return false;
+          });
+      if (iter_array != array->end())
+        msg_params.erase(*iter_set);
```

Comment:
@RomanReznichenkoLuxoft suggest adding `SDL_LOG_DEBUG` here and print the name of the key which was automatically erased from the message.

Reply:
Done